casr/src/bin/casr-lua.rs

```diff
@@ -0,0 +1,155 @@
+                .help("Stdin file for program"),
+        )
+        .arg(
+            Arg::new("timeout")
+                .short('t')
+                .long("timeout")
+                .action(ArgAction::Set)
+                .default_value("0")
+                .value_name("SECONDS")
+                .help("Timeout (in seconds) for target execution, 0 value means that timeout is disabled")
+                .value_parser(clap::value_parser!(u64))
+        )
+        .arg(
+            Arg::new("ignore")
+                .long("ignore")
+                .action(ArgAction::Set)
+                .value_parser(clap::value_parser!(PathBuf))
+                .value_name("FILE")
+                .help("File with regular expressions for functions and file paths that should be ignored"),
+        )
+        .arg(
+            Arg::new("strip-path")
+                .long("strip-path")
+                .env("CASR_STRIP_PATH")
+                .action(ArgAction::Set)
+                .value_name("PREFIX")
+                .help("Path prefix to strip from stacktrace"),
+        )
+        .arg(
+            Arg::new("ARGS")
+                .action(ArgAction::Set)
+                .num_args(1..)
+                .last(true)
+                .required(true)
+                .help("Add \"-- <path> <arguments>\" to run"),
+        )
+        .get_matches();
+
+    init_ignored_frames!("lua");
+    if let Some(path) = matches.get_one::<PathBuf>("ignore") {
+        util::add_custom_ignored_frames(path)?;
+    }
+    // Get program args.
+    let argv: Vec<&str> = if let Some(argvs) = matches.get_many::<String>("ARGS") {
+        argvs.map(|s| s.as_str()).collect()
+    } else {
+        bail!("Wrong arguments for starting program");
+    };
+
+    // Get stdin for target program.
+    let stdin_file = util::stdin_from_matches(&matches)?;
+
+    // Get timeout
+    let timeout = *matches.get_one::<u64>("timeout").unwrap();
+
+    // Run program.
+    let mut cmd = Command::new(argv[0]);
+    if let Some(ref file) = stdin_file {
+        cmd.stdin(std::fs::File::open(file)?);
+    }
+    if argv.len() > 1 {
+        cmd.args(&argv[1..]);
+    }
+    let result = util::get_output(&mut cmd, timeout, true)?;
+    let stderr = String::from_utf8_lossy(&result.stderr);
+
+    // Create report.
+    let mut report = CrashReport::new();
+    report.executable_path = argv[0].to_string();
+    if argv.len() > 1 {
+        if let Some(fname) = Path::new(argv[0]).file_name() {
+            let fname = fname.to_string_lossy();
+            if fname.starts_with("lua") && !fname.ends_with(".lua") && argv[1].ends_with(".lua") {
+                report.executable_path = argv[1].to_string();
+            }
+        }
+    }
+    report.proc_cmdline = argv.join(" ");
+    let _ = report.add_os_info();
+    let _ = report.add_proc_environ();
+
+    // Extract lua exception
+    let Some(exception) = LuaException::new(&stderr) else {
+        bail!("Lua exception is not found!");
```

Comment:
Lua code:

```lua
#!/bin/env lua

local function f1()
  error('error is triggered', 1)
end

local function f2()
  f1()
end

local function f3()
  f2()
end

local function f4()
  f3()
end

local function f5()
  f4()
end

local thread = coroutine.create(f5)
local executed_ok, message = coroutine.resume(thread)

if not executed_ok then
  print("Error: "..message.." in ".. debug.traceback(thread))
end
```

```
[0] ~/sources/casr $ lua exc2.lua 
Error: exc2.lua:4: error is triggered in stack traceback:
        [C]: in function 'error'
        exc2.lua:4: in function 'f1'
        exc2.lua:8: in function 'f2'
        exc2.lua:12: in function 'f3'
        exc2.lua:16: in function 'f4'
        exc2.lua:20: in function <exc2.lua:19>
[0] ~/sources/casr $ ./target/debug/casr-lua -o lua.casrep -- ./exc2.lua 
Error: Lua exception is not found!
[0] ~/sources/casr $ 
```

Should it be supported?

Reply:
Hm... I assumed there was always going to be something like `[C] in ?` or `[C] at 0xdeadbeaf` at the end of traceback, but apparently it's not

Reply:
Wait... That's just debug info in stdout, not error message (although it contains word `Error`)
Is it possible to rewrite this logging message to stderr?

Reply:
To sum up the above, I think we should not to support it